app/models/derived_variables/sales_log_variables.rb

```diff
@@ -160,6 +160,8 @@ def child_under_16_constraints!
     (start_index..6).each do |idx|
       if age_under_16?(idx)
         self["ecstat#{idx}"] = 9
+      elsif self["ecstat#{idx}"] == 9
+        self["ecstat#{idx}"] = nil
       end
     end
```

Comment:
I've added this so a user can change change the age of a person that was first set as under 16 to another age above 16. At the moment should they want to do that they would get an error:
_Answer cannot be over 16 as person’s 2 working situation is ‘child under 16‘._

As the working situation question for a child is not one they can access they'd get stuck. Now should they change the age e.g. to 17, the next question will be to choose the working situation.

Reply:
Can we add some tests for this to make sure it doesn't clear it when we don't want it to?